src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -377,6 +377,13 @@
   (mapv (partial str "DNS:")
         (utils/get-subject-dns-alt-names cert-or-csr)))
 
+(schema/defn subject-alt-names :- [schema/Str]
+  "Get the list of both DNS and IP alt names on the provided certificate or CSR.
+   Each name will be prepended with 'DNS:' or 'IP:'."
+  [cert-or-csr :- (schema/either Certificate CertificateRequest)]
+  (into (mapv (partial str "IP:") (utils/get-subject-ip-alt-names cert-or-csr))
+        (mapv (partial str "DNS:") (utils/get-subject-dns-alt-names cert-or-csr))))
```

Comment:
This is a WIP.  This would use my new "get-subject-ip-alt-names" method from jvm-ssl-utils to get the IP subject alt names and prepend them with "IP".

Reply:
Note the similarity of this method to the method "dns-alt-names" above, which only got the dns-alt names.